src/main/java/net/snowflake/client/core/SessionUtil.java

```diff
@@ -334,20 +328,75 @@ static SFLoginOutput openSession(
       }
     }
 
-    preNewSession(loginInput);
+    readCachedTokens(loginInput);
+
+    if (AccessTokenProviderFactory.isEligible(getAuthenticator(loginInput))) {
+      if (loginInput.getOauthAccessToken() != null) { // Access Token cached
+        loginInput.setAuthenticator(AuthenticatorType.OAUTH.name());
```

Comment:
will the new authenticators work with session renewal also?

Reply:
Yes, but I'm still thinking how to write automated test for it.